crates/eframe/src/native/run.rs

```diff
@@ -823,153 +1226,399 @@ mod glow_integration {
+                .as_ref()
+                .and_then(|r| r.glutin_ctx.borrow().window_maps.get(&id).copied())
+        }
+
+        fn viewport_id_from_window_id(&self, id: &WindowId) -> Option<ViewportId> {
+            self.running
+                .as_ref()
+                .and_then(|r| r.glutin_ctx.borrow().viewport_maps.get(id).copied())
         }
 
         fn save_and_destroy(&mut self) {
             if let Some(mut running) = self.running.take() {
                 crate::profile_function!();
-                running
-                    .integration
-                    .save(running.app.as_mut(), running.gl_window.window.as_ref());
+
+                running.integration.save(
+                    running.app.as_mut(),
+                    Some(
+                        &running
+                            .glutin_ctx
+                            .borrow()
+                            .window(ViewportId::ROOT)
+                            .borrow(),
+                    ),
+                );
                 running.app.on_exit(Some(&running.gl));
-                running.painter.destroy();
+                running.painter.borrow_mut().destroy();
             }
         }
 
-        fn run_ui_and_paint(&mut self) -> EventResult {
-            let Some(running) = &mut self.running else {
-                return EventResult::Wait;
-            };
-
-            if running.gl_window.window.is_none() {
+        fn run_ui_and_paint(&mut self, window_id: WindowId) -> EventResult {
+            if self.running.is_none() {
                 return EventResult::Wait;
             }
 
+            let Some(viewport_id) = self.viewport_id_from_window_id(&window_id) else {
+                return EventResult::Wait;
+            };
+
             #[cfg(feature = "puffin")]
             puffin::GlobalProfiler::lock().new_frame();
             crate::profile_scope!("frame");
 
-            let GlowWinitRunning {
-                gl_window,
-                gl,
-                app,
-                integration,
-                painter,
-            } = running;
-
-            let window = gl_window.window();
-
-            let screen_size_in_pixels: [u32; 2] = window.inner_size().into();
+            let (integration, app, glutin, painter, gl) = {
+                let running = self.running.as_mut().unwrap();
+                (
+                    &mut running.integration,
+                    &mut running.app,
+                    running.glutin_ctx.clone(),
+                    running.painter.clone(),
+                    running.gl.clone(),
+                )
+            };
 
-            egui_glow::painter::clear(
-                gl,
-                screen_size_in_pixels,
-                app.clear_color(&integration.egui_ctx.style().visuals),
-            );
+            // This will only happen if the viewport is sync
+            // That means that the viewport cannot be rendered by itself and needs his parent to be rendered
```

Comment:
What will only happen?

Reply:
The closure underneath that!

That request a repaint for the parent of that viewport!